addons/hr_holidays/models/hr_holidays.py

```diff
@@ -486,6 +483,17 @@ def action_validate(self):
                     leaves.action_validate()
         return True
 
+    def _validate_leave_request(self):
+        """ Validate the leave request (should be apply only on holiday_type='employee'
+            and holiday.type='remove').
```

Comment:
Maybe validate for that condition in the method, or remove that as precondition and filter `self` on the relevant leaves?